scripts/test-react.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+let fs = require("fs");
+let path = require("path");
+let { prepackSources } = require("../lib/prepack-node.js");
+let { ObjectCreate, CreateDataPropertyOrThrow, GetValue } = require("../lib/methods/index.js");
+let buildExpressionTemplate = require("../lib/utils/builder.js").default;
+let babel = require("babel-core");
+let React = require("react");
+let ReactTestRenderer = require("react-test-renderer");
+let { Value, AbstractValue, NativeFunctionValue, ObjectValue } = require("../lib/values/index.js");
+let { normalize } = require("../lib/utils/json.js");
+let { createMockReactComponent, createMockReactCloneElement } = require("../lib/react/mocks.js");
+/* eslint-disable no-undef */
+let { expect, describe, it } = global;
+
+let reactTestRoot = path.join(__dirname, "../test/react/");
+let prepackOptions = {
+  errorHandler: diag => "Fail",
+  internalDebug: true,
+  serialize: true,
+  uniqueSuffix: "",
+  maxStackDepth: 100,
+  reactEnabled: true,
+  additionalGlobals,
+  inlineExpressions: true,
+  omitInvariants: true,
+};
+
+function additionalGlobals(realm) {
```

Comment:
This helper relies on a lot of implementation details. Can you find a way to use the public API to express this?

The typical way you'd create something like this in the public API is by prepending a set up script like:

```js
require = __abstract('function', 'require');
```

Reply:
`additionalGlobals` is a public API option that I added in a previous PR for this exact purpose.

Reply:
I don't think we want the public API to expose an entire `realm` and all internals like `ObjectCreate` thought, right? We wouldn't want someone else to use it this way since these can break at any point.

Reply:
I thought we did want to expose these objects as `additionalGlobals` was meant as an escape hatch. I guess we could add an option `reactMocksEnabled` and then move this logic into core global instrincts?